yt_dlp/extractor/mediaset.py

```diff
@@ -160,17 +198,37 @@ def _parse_smil_formats(self, smil, smil_url, video_id, namespace=None, f4m_para
             video.attrib['src'] = re.sub(r'(https?://vod05)t(-mediaset-it\.akamaized\.net/.+?.mpd)\?.+', r'\1\2', video.attrib['src'])
         return super(MediasetIE, self)._parse_smil_formats(smil, smil_url, video_id, namespace, f4m_params, transform_rtmp_url)
 
+    def _check_drm_formats(self, tp_formats, video_id):
+        # get all manifest urls and remove duplicates
+        manifests = [f.get('manifest_url') for f in tp_formats]
+        manifests = list(dict.fromkeys(filter(None, manifests)))
+
+        # return original formats if not DRM manifest is found
+        if len(manifests) != 1 or (
+                len(manifests) == 1 and '_sampleaes/' not in manifests[0]):
+            return tp_formats
+
+        nodrm_manifest = re.sub(
+            r'_sampleaes/(\w+)_fp_', r'/\1_no_', manifests[0])
+
+        if nodrm_manifest == manifests[0]:
+            return tp_formats
+
+        return self._extract_m3u8_formats(
+            nodrm_manifest, video_id, m3u8_id='hls', fatal=False) or 'DRM'
```

Comment:
```py
    def _check_drm_formats(self, tp_formats, video_id):
        has_nondrm, drm_manifest = False, ''
        for f in tp_formats:
            if '_sampleaes/' in f.get('manifest_url'):
                drm_manifest = drm_manifest or f['manifest_url']
                f['has_drm'] = True
            if not f.get('has_drm'):
                has_nondrm = True

        nodrm_manifest = re.sub(r'_sampleaes/(\w+)_fp_', r'/\1_no_', drm_manifest)
        if has_nondrm or nodrm_manifest == drm_manifest:
            return

        tp_formats.extend(self._extract_m3u8_formats(
            nodrm_manifest, video_id, m3u8_id='hls', fatal=False) or [])
```
(untested)

Reply:
ok, fixed a couple of thing in your code to prevent TypeError